packages/mui-joy/src/Autocomplete/Autocomplete.test.tsx

```diff
@@ -2275,4 +2327,139 @@ describe('Joy <Autocomplete />', () => {
+        scrollToBottomCallCount: 1,
+      });
+    });
+
+    it('should call onScrollToBottom when scroll bar reaches bottom by keyboard', function test() {
+      if (/jsdom/.test(window.navigator.userAgent)) {
+        this.skip();
+      }
+      const onScrollToBottom = spy();
+
+      render(
+        <Autocomplete
+          open
+          autoFocus
+          options={['one', 'two', 'three', 'four', 'five']}
+          onScrollToBottom={onScrollToBottom}
+          slotProps={{
+            listbox: {
+              style: {
+                height: '100px',
+              },
+            },
+          }}
+        />,
+      );
+      testOnScrollToBottom({
+        reason: 'keyboard',
+        onScrollToBottom,
+        scrollToBottomCallCount: 1,
+      });
+    });
+
+    it('should call onScrollToBottom and custom listbox onScroll when provided', function test() {
+      if (/jsdom/.test(window.navigator.userAgent)) {
+        this.skip();
+      }
+      const onScrollToBottom = spy();
+      const onScroll = spy();
+
+      render(
+        <Autocomplete
+          open
+          options={['one', 'two', 'three', 'four', 'five']}
+          onScrollToBottom={onScrollToBottom}
+          autoFocus
+          slotProps={{
+            listbox: {
+              style: {
+                height: '100px',
+              },
+              onScroll,
+            },
+          }}
+        />,
+      );
+      testOnScrollToBottom({
+        reason: 'keyboard',
+        onScrollToBottom,
+        scrollToBottomCallCount: 1,
+      });
+      testOnScrollToBottom({
+        reason: 'mouse',
+        onScrollToBottom,
+        scrollToBottomCallCount: 2,
+      });
+      expect(onScroll.callCount).to.equal(2);
+    });
+
+    it("should not call onScrollToBottom when listbox doesn't have scroll bar", function test() {
+      if (/jsdom/.test(window.navigator.userAgent)) {
+        this.skip();
+      }
+      const onScrollToBottom = spy();
+
+      render(
+        <Autocomplete
+          autoFocus
+          open
+          options={['one', 'two', 'three', 'four', 'five']}
+          onScrollToBottom={onScrollToBottom}
+        />,
+      );
+      testOnScrollToBottom({
+        reason: 'keyboard',
+        onScrollToBottom,
+        scrollToBottomCallCount: 0,
+      });
+      testOnScrollToBottom({
+        reason: 'mouse',
+        onScrollToBottom,
+        scrollToBottomCallCount: 0,
+      });
+    });
+  });
+
+  describe('prop: showLoadingWithOptions', () => {
+    it('should display loading text with options', () => {
+      render(
+        <Autocomplete loading showLoadingWithOptions open options={['one', 'two', 'three']} />,
+      );
+      expect(document.querySelector(`.${classes.loading}`)?.textContent).to.equal('Loading…');
+      expect(screen.getAllByRole('option').length).to.equal(3);
+    });
+
+    it('should not display options with loading text when showLoadingWithOptions is false', () => {
```

Comment:
I didn't understand this text. If `options` are not provided (an empty array `[]`), the `options` will not be displayed anyway, right?

Reply:
ah yes sorry, this test is not required. removed it